### ISSUE TITLE
Segfault on (Query () Goals) with sertop 8.15.0+0.15.3

### ISSUE THREAD

**Issue by rwhender** [ejgallego/coq-serapi#331]
I'm trying to to use `sertop` to interact with a Coq file in the [Coq-HoTT](https://github.com/HoTT/Coq-HoTT) project. I am having trouble parsing [this particular version of this file](https://github.com/HoTT/Coq-HoTT/blob/fe4cbfb6c541139e95e9650005014e736668117f/theories/Basics/Contractible.v).

My general process involves parsing each line of a file and then querying goals as part of a heuristic process to determine whether the line is part of a proof or not. The following conditions lead to a segfault in `sertop`:

- I am using Coq 8.15.2.
- The project builds with no errors using the project's build instructions.
- From the root of the project repository, after building, I run `rlwrap sertop --omit_att --omit_loc --printer=human --no_prelude -Q contrib,HoTT.Contrib -R theories,HoTT --indices-matter --implicit`.
  - Note that for this project, `--no-prelude` is required, and this requirement seems to preclude us from using `sercomp` to try to recreate the bug with less effort.
- I then step through the previously linked file, using `(Add () "<sentence>")` and `(Exec <state_id>)` to run each sentence.
- After executing [line 28](https://github.com/HoTT/Coq-HoTT/blob/fe4cbfb6c541139e95e9650005014e736668117f/theories/Basics/Contractible.v#L28), I run `(Query () Goals)` and get the following output:

```
(Add () "exists ((contr x)^ @ contr y).")                                                                                                                                                                        (Answer 25 Ack)
(Answer 25 (Added 16 [LOC] NewAddTip))
(Answer 25 Completed)
(Exec 16)
(Answer 26 Ack)
(Feedback
 ((doc_id 0) (span_id 16) (route 0) (contents (ProcessingIn master))))
(Feedback ((doc_id 0) (span_id 15) (route 0) (contents Processed)))
(Feedback ((doc_id 0) (span_id 16) (route 0) (contents Processed)))
(Answer 26 Completed)
(Query () Goals)
(Answer 27 Ack)
rlwrap: warning: sertop crashed, killed by SIGSEGV (core dumped).
rlwrap itself has not crashed, but for transparency,
it will now kill itself with the same signal

                                                                                                                                                                                                                 warnings can be silenced by the --no-warnings (-n) option
Segmentation fault (core dumped)
```

(Just a note that I am not affiliated with Coq-HoTT).

**Comment by rwhender** [comment]
Thanks for the response.

I am using coq-serapi 8.15.0+0.15.3, coq 8.15.2, and OCaml 4.09.1.

That is a strange complication. If I run your script, I do not observe the segfault.

However, I was able to find a more concise way of reproducing the error. If I create a new file (say "theories/Basics/Contracitble_truncated.v") that is truncated after line 28, then follow your script (except using `(Query () Goals)`), I observe a segfault.

```
(ReadFile "theories/Basics/Contractible_truncated.v")
(Exec 16)
(Query () Goals)
```

I'll note here that I did rebuild by running `make clean` then `make` once again to make sure no strange artifacts were causing problems. The segfault still occurred in this case.

A note about our application here might be useful to explain this potentially odd-looking approach. We wish to know as much as possible about what Coq knows about the current state of goals, proof mode, etc. for every sentence in a Coq source file. We use the feedback from the `(Add...` command to determine the state that needs to be executed, then execute it, then query the goals after every execution to determine the change in proof state (if any).
